hyperspy/tests/signal/test_tools.py

```diff
@@ -528,3 +528,25 @@ def test_histogram_axis_changes(self):
         assert_array_equal(h1.data, h2.data)
         nt.assert_equal(h1.axes_manager[-1].size,
                         h2.axes_manager[-1].size,)
+
+    def test_masked_array_mean(self):
+        s = self.s
+        mask = (s.data > 0.5)
+        s.data = np.ones_like(s.data)
```

Comment:
I would rather you used something like `np.arange(s.data.size).reshape(s.data.shape)`. When all elements in an array are 1, testing if a particular subset has a mean of 1 is not very indicative that the subset was selected correctly...